pkg/gatherers/clusterconfig/operators.go

```diff
@@ -142,61 +153,66 @@ func collectClusterOperatorResources(ctx context.Context,
 	}
 	var res []clusterOperatorResource
 	for _, ro := range relObj {
-		key := fmt.Sprintf("%s-%s", ro.Group, strings.ToLower(ro.Resource))
-		versions := resVer[key]
-		for _, v := range versions {
-			gvr := schema.GroupVersionResource{Group: ro.Group, Version: v, Resource: strings.ToLower(ro.Resource)}
-			clusterResource, err := dynamicClient.Resource(gvr).Get(ctx, ro.Name, metav1.GetOptions{})
-			if err != nil {
-				klog.V(2).Infof("Unable to list %s resource due to: %s", gvr, err)
-			}
-			if clusterResource == nil {
-				continue
-			}
-			kind, err := utils.NestedStringWrapper(clusterResource.Object, "kind")
-			if err != nil {
-				continue
-			}
-			apiVersion, err := utils.NestedStringWrapper(clusterResource.Object, "apiVersion")
-			if err != nil {
-				continue
-			}
-			name, err := utils.NestedStringWrapper(clusterResource.Object, "metadata", "name")
-			if err != nil {
-				continue
-			}
-			spec, ok := clusterResource.Object["spec"]
-			if !ok {
-				klog.Warningf("Can't find spec for cluster operator resource %s", name)
-			}
-			anonymizeIdentityProviders(clusterResource.Object)
-			res = append(res, clusterOperatorResource{Spec: spec, Kind: kind, Name: name, APIVersion: apiVersion})
+		gr := schema.GroupResource{
+			Group:    ro.Group,
+			Resource: ro.Resource,
 		}
+		version := resVer[gr]
+		gvr := gr.WithVersion(version)
+		clusterResource, err := getRelatedObjectResource(ctx, dynamicClient, ro, co.Name, gvr)
+		if err != nil {
+			klog.V(2).Infof("Unable to get %s resource due to: %s", fmt.Sprintf("%s.%s", ro.Resource, ro.Group), err)
+			continue
+		}
+		spec, ok := clusterResource.Object["spec"]
```

Comment:
When I use this way of checking if a key exists in a map, I usually prefer to rename the second variable `exists` rather than `ok`. That way next line will display `if !exists {` what I find more readable.

However I saw this check several times here and on other repos. Out of curiosity, is this some kind of convention?

Reply:
BTW, not a change request or something like that. Just asking to feed my knowledge daemon.

Reply:
Thanks. I don't have strong opinion about this to be honest. I don't think it's really a convention. We can go or have `exists` as well.

Reply:
I think that both works fine. AFAIK, in Go, the most common is to use `ok`, but not a strong opinion, if someone use `exists` instead of `ok` I'll be fine with it as well, once the verification happens in the following line, it is easy to read anyway.